### PR TITLE
Updating/improving/creating docstring codebase

### PR DESCRIPTION
**Pull request by basfroman**
The changes contain changes/additions, as well as supplemented and updated examples of code usage.
The approaches to describing type annotations have been changed to bring the code base to consistency.
Other minor changes and refactoring.

DO NOT MERGE without @rajkaramchedu approval

### CODE REVIEW

File: bittensor/core/axon.py [hunk 1]
```diff
@@ -98,6 +97,16 @@ class FastAPIThreadedServer(uvicorn.Server):
...
 
+    Example Usage::
+
+        self.app = FastAPI()
+        log_level = "trace"
+        self.fast_config = uvicorn.Config(self.app, host="0.0.0.0", port=self.config.axon.port, log_level=log_level)
+        self.fast_server = FastAPIThreadedServer(config=self.fast_config)
+        self.fast_server.start()
+        # do something
+        self.fast_server.stop()
```
Comment (thewhaleking) [inline bittensor/core/axon.py:108]: Example usage is typically written preceded by `>>>`

File: bittensor/core/axon.py [hunk 2]
```diff
@@ -163,7 +172,7 @@ def stop(self):
 
 class Axon:
     """
-    The ``axon`` class in Bittensor is a fundamental component that serves as the server-side interface for a neuron within the Bittensor network.
+    The ``Axon`` class in Bittensor is a fundamental component that serves as the server-side interface for a neuron within the Bittensor network.
```
Comment (thewhaleking) [inline bittensor/core/axon.py:175]: Only needs a single backtick. Not sure where our style of double-backticks comes from

File: bittensor/core/metagraph.py [hunk 3]
```diff
@@ -867,16 +880,21 @@ def __init__(
     ):
         """
         Initializes a new instance of the metagraph object, setting up the basic structure and parameters based on the provided arguments.
-        This method is the entry point for creating a metagraph object,
-        which is a central component in representing the state of the Bittensor network.
+        This class require installed Torch.
```
Comment (thewhaleking) [inline bittensor/core/metagraph.py]: requires*

This is the only blocker.

File: bittensor/utils/balance.py [hunk 4]
```diff
@@ -230,9 +230,9 @@ def __abs__(self):
     @staticmethod
     def from_float(amount: float):
         """
-        Given tao (float), return Balance object with rao(int) and tao(float), where rao = int(tao*pow(10,9))
+        Given tao, return :func:`Balance` object with rao(``int``) and tao(``float``), where rao = int(tao*pow(10,9))
```
Comment (thewhaleking) [inline bittensor/utils/balance.py:233]: is the use of the quotes inside a docstring really necessary?

**Review by thewhaleking — CHANGES_REQUESTED** [review]

File: bittensor/core/axon.py (hunk 1 above)
Comment (basfroman) [reply bittensor/core/axon.py:108]: I thought so too. It turns out the magic of `::` and indentation is enough for `sphinxcontrib-napoleon` to create a code snippet in the documentation.

File: bittensor/core/axon.py (hunk 2 above)
Comment (basfroman) [reply bittensor/core/axon.py:175]: As I understand it, single backtick is used in cases of link and using the keyword `:func:` (For example: :func:`uvicorn.Server`). And double backticks allow to highlight a word.

File: bittensor/utils/balance.py (hunk 4 above)
Comment (basfroman) [reply bittensor/utils/balance.py:233]: I would delete. Raj wants to highlight the types of variables. Although they are described below.

File: bittensor/core/metagraph.py (hunk 3 above)
Comment (basfroman) [reply bittensor/core/metagraph.py]: fixed

File: bittensor/utils/balance.py (hunk 4 above)
Comment (thewhaleking) [reply bittensor/utils/balance.py:233]: I get that. I mean the `"Balance"` vs `Balance`

**Review by thewhaleking — APPROVED** [review]

File: bittensor/utils/balance.py (hunk 4 above)
Comment (basfroman) [reply bittensor/utils/balance.py:233]: I see. `:func:''Balance''` creates anchor link as `#` in HTML. I did it based on what I've seen Raj do. He will ultimately do it the way he wants users to see it :)

**Review by rajkaramchedu — APPROVED** [review]

**Pull request merged**